crates/oxc_linter/src/rules/react/no_children_prop.rs

```diff
@@ -0,0 +1,184 @@
+use oxc_ast::{
+    ast::{
+        Argument, CallExpression, Expression, JSXAttributeItem, JSXAttributeName,
+        ObjectPropertyKind,
+    },
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-react/no-children-prop: Avoid passing children using a prop.")]
+#[diagnostic(
+    severity(warning),
+    help("The canonical way to pass children in React is to use JSX elements")
+)]
+struct NoChildrenPropDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoChildrenProp;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Children should always be actual children, not passed in as a prop.
+    ///
+    /// When using JSX, the children should be nested between the opening and closing tags.
+    ///
+    /// When not using JSX, the children should be passed as additional arguments to `React.createElement`.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <div children='Children' />
+    ///
+    /// <MyComponent children={<AnotherComponent />} />
+    /// <MyComponent children={['Child 1', 'Child 2']} />
+    /// React.createElement("div", { children: 'Children' })
+    ///
+    /// // Good
+    ///
+    /// <div>Children</div>
+    /// <MyComponent>Children</MyComponent>
+    ///
+    /// <MyComponent>
+    ///   <span>Child 1</span>
+    ///   <span>Child 2</span>
+    /// </MyComponent>
+    ///
+    /// React.createElement("div", {}, 'Children')
+    /// React.createElement("div", 'Child 1', 'Child 2')
+    ///
+    ///
+    /// ```
+    NoChildrenProp,
+    correctness
+);
+
+impl Rule for NoChildrenProp {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::JSXElement(jsx_elem) => {
+                if let Some(span) = jsx_elem.opening_element.attributes.iter().find_map(|attr| {
+                    let JSXAttributeItem::Attribute(attr) = attr else {
+                        return None;
+                    };
+
+                    let JSXAttributeName::Identifier(attr_ident) = &attr.name else {
+                        return None;
+                    };
+                    if attr_ident.name == "children" {
+                        Some(attr.span)
+                    } else {
+                        None
+                    }
+                }) {
+                    ctx.diagnostic(NoChildrenPropDiagnostic(span));
+                }
+            }
```

Comment:
You are free to add `AstKind` for the missing nodes

https://github.com/web-infra-dev/oxc/blob/c95f2e093742290343efcd2120b8359d0b58bb05/crates/oxc_ast/src/visit.rs#L1038-L1051

I didn't add them not because they are not needed, it was simply they weren't used anywhere yet.